apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.ts

```diff
@@ -0,0 +1,375 @@
+    styleUrls: ['./purchase-order-preview.component.scss']
+})
+
+export class PurchaseOrderPreviewComponent implements OnInit, OnChanges, OnDestroy, AfterViewInit {
+    /* Taking input of all purchase orders */
+    @Input() public purchaseOrders: any;
+    /* Taking input of company unique name */
+    @Input() public companyUniqueName: any;
+    /* Taking input of purchase order unique name */
+    @Input() public purchaseOrderUniqueName: any;
+    /* Search element */
+    @ViewChild('searchElement') public searchElement: ElementRef;
+    /* Confirm box */
+    @ViewChild('poConfirmationModel') public poConfirmationModel: ModalDirective;
+    /* Modal instance */
+    public modalRef: BsModalRef;
+    /* This will hold state of activity history aside pan */
+    public activityHistoryAsideState: string = 'out';
+    /* This will hold purchase order data */
+    public purchaseOrder: any = {};
+    /* Send email request params object */
+    public sendEmailRequest: any = {};
+    /* This will hold if it's loading or not */
+    public isLoading: boolean = false;
+    /* This will hold giddh date format */
+    public giddhDateFormat: any = GIDDH_DATE_FORMAT_UI;
+    /* This will hold inventory settings */
+    public inventorySettings: any;
+    /* Observable to unsubscribe all the store listeners to avoid memory leaks */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /* This will hold the filtered orders */
+    public filteredData: any[] = [];
+    /* This will hold if we need to show GST */
+    public showGSTINNo: boolean;
+    /* This will hold if we need to show TRN */
+    public showTRNNo: boolean;
+    /* This will hold selected company details */
+    public selectedCompany: any;
+    /* This will hold list of vat supported countries */
+    public vatSupportedCountries = VAT_SUPPORTED_COUNTRIES;
+    /* This will hold form fields */
+    public formFields: any[] = [];
+    /* True, if the Giddh supports the taxation of the country (not supported now: UK, US, Nepal, Australia) */
+    public shouldShowTrnGstField: boolean = false;
+    /* Onboarding params object */
+    public onboardingFormRequest: OnboardingFormRequest = { formName: 'onboarding', country: '' };
+
+    constructor(private store: Store<AppState>, private modalService: BsModalService, public purchaseOrderService: PurchaseOrderService, private toaster: ToasterService, public router: Router, private commonActions: CommonActions, private invoiceActions: InvoiceActions) {
+        this.getInventorySettings();
+        this.store.dispatch(this.invoiceActions.getInvoiceSetting());
+    }
+
+    /**
+     * Initializes the component
+     *
+     * @memberof PurchaseOrderPreviewComponent
+     */
+    public ngOnInit(): void {
+        this.getPurchaseOrder();
+
+        if (this.purchaseOrders && this.purchaseOrders.items) {
+            this.filteredData = this.purchaseOrders.items;
+        }
+
+        this.store.pipe(select(state => {
```

Comment:
Avoid listening to whole app state as it will be fired on every state change even if it is not related to companies.
Also use takeUntil on subcription